### PR TITLE
Add category service types

### PR DESCRIPTION
**Pull request by sorja**

### CODE REVIEW

**Review by SteRiccio — APPROVED** [review]

File: src/category/importSummary.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  levelIndex: number
+  levelName: string
+  name: string
+  type: ImportColumnType
+}
+
+export interface ImportSummary {
+  columns: { [key: string]: ImportSummaryColumn }
+  filePath: string
+}
```
Comment (minotogna) [inline src/category/importSummary.ts:23]: add Category prefix

File: src/category/service.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import { User } from 'src/auth'
+import { Category } from './category'
+import { CategoryLevel } from './level'
+import { CategoryItem } from './item'
+import { ImportSummary } from './importSummary'
+
+export interface CategoryImportJob {
+  // Todo
+}
```
Comment (minotogna) [inline src/category/service.ts:9]: you don't need this. 
use JobStatus

File: src/category/service.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
```
Comment (minotogna) [inline src/category/service.ts:10]: let's split into 3. CategoryService, CategoryLevelService, CategoryItemService.
so we avoid prefix like level, item to method names.

Comment (minotogna) [inline src/category/service.ts:10]: surveyId is number anywhere used

File: src/category/service.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    surveyId: string
+    user: User
+  }): Promise<{ item: CategoryItem; category: Category }>
+
+  // ==== READ
+  count(options: { draft: boolean; user: User }): Promise<number>
+
+  getMany(options: {
+    draft: boolean
+    includeValidation: boolean
```
Comment (minotogna) [inline src/category/service.ts:42]: call this validate

File: src/category/service.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    limit: number
+    offset: number
+    search: string
```
Comment (minotogna) [inline src/category/service.ts:45]: draft, validate, limit, offset, search, are all optional

File: src/category/service.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    surveyId: string
+  }): Promise<Array<Category>>
+
+  get(options: { validate: boolean; draft: boolean; categoryUuid: string; surveyId: string }): Promise<Category>
```
Comment (minotogna) [inline src/category/service.ts:49]: draft, validate are optional

File: src/category/service.ts [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+  getItemsByParentUuid(options: {
```
Comment (minotogna) [inline src/category/service.ts:51]: this will be just getMany in CategoryItemService

File: src/category/service.ts [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+  }): Promise<{ level: CategoryLevel; category: Category }>
+
+  updateItemProp(options: {
+    categoryUuid: string
+    itemUuid: string
+    key: string
+    surveyId: string
+    user: User
+    value: any
+  }): Promise<{ item: CategoryItem; category: Category }>
```
Comment (minotogna) [inline src/category/service.ts:84]: as we did for other service, do not call methods updateProps, just update and pass the relative Props object

**Review by minotogna — CHANGES_REQUESTED** [review]

File: src/category/categoryItermService.ts [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    draft?: boolean
+    parentUuid?: string
+    surveyId: number
+  }): Promise<Array<CategoryItem>>
+
+  // ==== UPDATE
+  update(options: {
+    categoryUuid: string
+    itemUuid: string
+    key: string
```
Comment (minotogna) [inline src/category/categoryItermService.ts:28]: do we still need key here ?

File: src/category/categoryItermService.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    surveyId: number
+    user: User
+    props: CategoryItemProps
+  }): Promise<{ item: CategoryItem; category: Category }>
+
+  // ==== DELETE
+  deleteItem(options: { itemUuid: string; categoryUuid: string; surveyId: number; user: User }): Promise<Category>
```
Comment (minotogna) [inline src/category/categoryItermService.ts:35]: call this delete

File: src/category/categoryLevelService.ts [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    search?: string
+    surveyId: number
+  }): Promise<Array<CategoryLevel>>
+
+  get(options: { validate?: boolean; draft?: boolean; categoryUuid: string; surveyId: number }): Promise<CategoryLevel>
+
+  // ==== UPDATE
+  update(options: {
+    categoryUuid: string
+    key: string
```
Comment (minotogna) [inline src/category/categoryLevelService.ts:25]: do we still need key here ?

File: src/category/index.ts [hunk 12]
```diff
@@ -1,5 +1,9 @@
-export type { Category, CategoryItemExtraDef, CategoryItemExtraDefs, CategoryProps } from './category'
+export type { Category, CategoryItemExtraDef, CategoryProps } from './category'
 
 export type { CategoryItem, CategoryItemProps } from './item'
 
-export type { CategoryLevel, CategoryLevelProps, CategoryLevels } from './level'
+export type { CategoryLevel, CategoryLevelProps } from './level'
+
+export type { CategoryImportSummary, CategoryImportSummaryColumn } from './categoryImportSummary'
+
```
Comment (minotogna) [inline src/category/index.ts:8]: we need to export the service from index

File: src/category/categoryLevelService.ts [hunk 13]
```diff
@@ -0,0 +1,34 @@
+import { User } from 'src/auth'
```
Comment (minotogna) [inline src/category/categoryLevelService.ts:1]: call the file levelService

File: src/category/categoryItermService.ts [hunk 14]
```diff
@@ -0,0 +1,36 @@
+import { User } from 'src/auth'
```
Comment (minotogna) [inline src/category/categoryItermService.ts:1]: call the file itemService

File: src/category/service.ts [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+    limit?: number
+    offset?: number
+    search?: string
+    surveyId: number
+  }): Promise<Array<Category>>
+
+  // ==== UPDATE
+  update(options: {
+    categoryUuid: string
+    key: string
```
Comment (minotogna) [inline src/category/service.ts:36]: do we still need key here?

**Review by minotogna — CHANGES_REQUESTED** [review]

**Review by SteRiccio — APPROVED** [review]

**Review by minotogna — APPROVED** [review]

**Pull request merged**